### PR TITLE
Custom mouse and key actions through config

### PR DESCRIPTION
**Pull request by ZacZhangzhuo**
This PR will alow `Custom mouse and key actions through config`. Users can define their keys in the config file.

### What type of change is this?

-   Add example `example_custom_keys.py` in the examples/control.
-   Action class which controls all the key actions.

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_view2/app/controller.py [hunk 1]
```diff
@@ -33,11 +33,77 @@ def __init__(self, app, controller_config: Dict):
         self.keys = controller_config["actions"]["keys"]
         self.mouse = Mouse()
         self.key_status = {}
-        self.actions = {}  # Currently, no customized actions supported.
```
Comment (ZacZhangzhuo) [inline src/compas_view2/app/controller.py:36]: now supported

**Comment by ZacZhangzhuo** [comment]
Would this be the last PR for `compas_view2`? @Licini

File: src/compas_view2/app/controller.py [hunk 2]
```diff
@@ -33,11 +33,77 @@ def __init__(self, app, controller_config: Dict):
...
+        self.actions = self.keys.keys()
         self.supported_keys = {
```
Comment (ZacZhangzhuo) [inline src/compas_view2/app/controller.py:37]: This variable ` self.supported_keys` is a bit long, but should be fine for this version

**Comment by Licini** [comment]
> Would this be the last PR for `compas_view2`? @Licini

Yes 😎

File: docs/examples/control/example_custom_keys.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            },
+        },
+    }
+}
+
+viewer = App(config=config)
+
+curve = Bezier([[0, 0, 0], [3, 6, 0], [5, -3, 0], [10, 0, 0]])
+pointobj = viewer.add(Point(*curve.point(0)), pointsize=20, pointcolor=(1, 0, 0))
+curveobj = viewer.add(Polyline(curve.locus()), linewidth=2)
```
Comment (Licini) [inline docs/examples/control/example_custom_keys.py:30]: can we change this part of example to something that is easy to click on? maybe bunch of boxes

File: src/compas_view2/app/controller.py [hunk 4]
```diff
@@ -33,11 +33,77 @@ def __init__(self, app, controller_config: Dict):
...
             "shift": QtCore.Qt.Key_Shift,
             "control": QtCore.Qt.Key_Control,
             "alt": QtCore.Qt.Key_Alt,
+            "space": QtCore.Qt.Key_Space,
```
Comment (Licini) [inline src/compas_view2/app/controller.py:41]: we should mention somewhere in documentation all these supported keys

File: docs/examples/control/example_custom_keys.py (hunk 3 above)
Comment (ZacZhangzhuo) [reply docs/examples/control/example_custom_keys.py]: Done
image

**Pull request merged**